attest/attest.go

```diff
@@ -97,12 +98,19 @@ const (
 	keyEncodingParameterized
 )
 
+// Handle represents different types of a key handle.
+type Handle struct {
+	Native *tpmutil.Handle
```

Comment:
I think this should either be an opaque handle with private fields, or just use an empty interface{}.

E.g. this should either be

type Handle struct {
    // contains private fields
}

Does this need to interface with other code or can it just be local to go-attestation? Basically does anyone else ever need to import a handle?

How do you expect to actually call Certify(), for example?

Reply:
It is needed for the case when a key created outside go-attestation has to be certified. For instance, a win pcp `key` is created with `certtostore` and we would like to get it certified by our `attestClient` (with `ak *AK`). Then I expect to call something like `attestClient.Certify(Handle{PCP:key.Handle()})` that would call `ak.Certify(..)`. 

I'm equally ok with the current `Handle` or
```
type Handle struct {
   Value interface{}
}
```

Reply:
Thanks for the info. The Handle suggestion was for type safety. If that's the case, we should either have constructors for Handle:

```
type Handle struct {
    // private field
}
func TLS12Handle(h uintptr) *Handle {
    // ...
}
func TLS20Handle(h tpmutil.Handle) *Handle {
    // ...
}
```

Or just have Certify take the handle then type switch (which is what you did originally):

```
// For  a TPM 1.2 handle pass a uintptr, for a TPM 2.0 handle pass a tpmutil.Handle.
func (...) Certify(handle interface{}) ... {

}
```

Given that I think I'd prefer the latter? Just do the type switch.

Reply:
Done. (I do type checking in the underlying ak.Certify()).